### DISCUSSION TITLE
Cast widgets

### DISCUSSION THREAD

**Discussion by abau** [fltk-rs/fltk-rs#1172] [Q&A] [answered]
Is there a way to safely cast to widget types other than groups (as described [here](https://github.com/fltk-rs/fltk-rs/issues/809))? 

For example, I would like to know if a given `Widget` implements `MenuExt`.

**Comment by MoAlyousef** [comment, marked as answer]
Unfortunately FLTK doesn’t offer such a method as as_menu (or anything other than as_group and as_window). 
You can keep a reference alive for the widgets that implement MenuExt (by using something like a static HashMap), or by casting a type-erased widget into a Menu or Choice (requires unsafe).

**Reply by abau** [reply]
I see. Thanks for your answer!
